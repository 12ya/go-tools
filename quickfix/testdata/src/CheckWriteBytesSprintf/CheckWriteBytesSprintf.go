package pkg

import (
	"bytes"
	"fmt"
	"io"
)

func _(w io.Writer) {
	w.Write([]byte(fmt.Sprint("abc", "de")))   // want `should use fmt.Fprint`
	w.Write([]byte(fmt.Sprintf("%T", w)))      // want `should use fmt.Fprintf`
	w.Write([]byte(fmt.Sprintln("abc", "de"))) // want `should use fmt.Fprintln`
}

func fn() {
	buf := new(bytes.Buffer)
	var sw io.StringWriter

	buf.WriteString(fmt.Sprint("abc", "de"))   // want `should use fmt.Fprint`
	buf.WriteString(fmt.Sprintf("%T", 0))      // want `should use fmt.Fprintf`
	buf.WriteString(fmt.Sprintln("abc", "de")) // want `should use fmt.Fprintln`

	// We can't suggest fmt.Fprint here. We don't know if sw implements io.Writer.
	sw.WriteString(fmt.Sprint("abc", "de"))
	sw.WriteString(fmt.Sprintf("%T", 0))
	sw.WriteString(fmt.Sprintln("abc", "de"))
}
