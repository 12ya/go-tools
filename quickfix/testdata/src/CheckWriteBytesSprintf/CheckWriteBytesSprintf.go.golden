package pkg

import (
	"bytes"
	"fmt"
	"io"
)

func _(w io.Writer) {
	fmt.Fprint(w, "abc", "de")   // want `should use fmt.Fprint`
	fmt.Fprintf(w, "%T", w)      // want `should use fmt.Fprintf`
	fmt.Fprintln(w, "abc", "de") // want `should use fmt.Fprintln`
}

func fn() {
	buf := new(bytes.Buffer)
	var sw io.StringWriter

	fmt.Fprint(buf, "abc", "de")   // want `should use fmt.Fprint`
	fmt.Fprintf(buf, "%T", 0)      // want `should use fmt.Fprintf`
	fmt.Fprintln(buf, "abc", "de") // want `should use fmt.Fprintln`

	// We can't suggest fmt.Fprint here. We don't know if sw implements io.Writer.
	sw.WriteString(fmt.Sprint("abc", "de"))
	sw.WriteString(fmt.Sprintf("%T", 0))
	sw.WriteString(fmt.Sprintln("abc", "de"))
}
